src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java

```diff
@@ -0,0 +1,442 @@
+        if (rack.isEmpty() && (rack.getOtherChest() == null || rack.getOtherChest().isEmpty()))
+        {
+            if(rack.getOtherChest() != null)
+            {
+                if(rack.isMain())
+                {
+                    return state.withProperty(BlockMinecoloniesRack.VARIANT, BlockMinecoloniesRack.EnumType.DEFAULTDOUBLE)
+                            .withProperty(FACING, BlockPosUtil.getFacing(rack.getNeighbor(), pos));
+                }
+                else
+                {
+                    return state.withProperty(BlockMinecoloniesRack.VARIANT, EnumType.EMPTYAIR);
+                }
+            }
+            else
+            {
+                return state.withProperty(BlockMinecoloniesRack.VARIANT, EnumType.DEFAULT);
+            }
+        }
+        else
+        {
+            if(rack.getOtherChest() != null)
+            {
+                if(rack.isMain())
+                {
+                    return state.withProperty(BlockMinecoloniesRack.VARIANT, EnumType.FULLDOUBLE)
+                            .withProperty(FACING, BlockPosUtil.getFacing(rack.getNeighbor(), pos));
+                }
+                else
+                {
+                    return state.withProperty(BlockMinecoloniesRack.VARIANT, BlockMinecoloniesRack.EnumType.EMPTYAIR);
+                }
+            }
+            else
+            {
+                return state.withProperty(BlockMinecoloniesRack.VARIANT, EnumType.FULL);
+            }
+        }
+    }
+
+    /**
+     * Check if a certain block should be replaced with a rack.
+     * @param block the block to check.
+     * @return true if so.
+     */
+    public static boolean shouldBlockBeReplacedWithRack(final Block block)
+    {
+        return block == Blocks.CHEST;
+    }
+
+    @Override
+    public int getMetaFromState(@NotNull final IBlockState state)
+    {
+        return state.getValue(FACING).getHorizontalIndex();
+    }
+
+    @Override
+    public void neighborChanged(final IBlockState state, final World worldIn, final BlockPos pos, final Block blockIn, final BlockPos fromPos)
+    {
+        if (state.getBlock() instanceof BlockMinecoloniesRack)
+        {
+            final TileEntity rack = worldIn.getTileEntity(pos);
+            for (final EnumFacing offsetFacing : BlockHorizontal.FACING.getAllowedValues())
+            {
+                final BlockPos neighbor = pos.offset(offsetFacing);
+                final Block block = worldIn.getBlockState(neighbor).getBlock();
+                if (rack instanceof TileEntityRack && pos.getY() == neighbor.getY() && !pos.equals(neighbor) && !pos.equals(BlockPos.ORIGIN)
+                        && (block instanceof BlockMinecoloniesRack || blockIn instanceof BlockMinecoloniesRack))
+                {
+                    ((TileEntityRack) rack).neighborChanged(neighbor);
+                }
+            }
+        }
+        super.neighborChanged(state, worldIn, pos, blockIn, fromPos);    }
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this closing curly brace to the next line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)